src/player_compevents.c

```diff
@@ -497,6 +502,74 @@ long computer_event_check_rooms_full(struct Computer2 *comp, struct ComputerEven
     return ret;
 }
 
+long computer_event_attack_door(struct Computer2* comp, struct ComputerEvent* cevent, struct Event* event)
+{
+    SYNCDBG(18, "Starting for %s", cevent->name);
+    struct Thing* thing = thing_get(event->target);
+    if (!thing_is_deployed_door(thing))
+    {
+        SYNCDBG(8, "Target %s is not a door", thing_model_name(thing));
+        return 0;
+    }
+    if (!players_are_enemies(comp->dungeon->owner, thing->owner))
+    {
+        SYNCDBG(8, "Door owner is no longer an enemy");
+        return 0;
+    }
+
+    struct Coord3d doorpos = thing->mappos;
+
+    struct Coord3d freepos;
+    if (!get_computer_drop_position_near_subtile(&freepos, comp->dungeon, coord_subtile(event->mappos_x), coord_subtile(event->mappos_y))) {
+        SYNCDBG(8, "No drop position near (%d,%d) for %s", (int)coord_subtile(event->mappos_x), (int)coord_subtile(event->mappos_y), cevent->name);
+        return 0;
+    }
+
+    if (!computer_find_non_solid_block(comp, &freepos)) {
+        SYNCDBG(8, "Drop position is solid for %s", cevent->name);
+        return 0;
+    }
+
+    if (computer_able_to_use_power(comp, PwrK_HAND, 1, 1))
```

Comment:
a)AI in emergency mode should not attack on sight
b)AI should `compuchat` about his desire to break door
c)AI could choose weak creature as player do

Reply:
A) I'll look into this
B) OK
C) Wouldn't that massively delay the AI on stuff like magic doors?

Reply:
A) Emergency mode is low on money, so he should explore in those cases
B) Done
C) I think the balance of which units it picks up is fine as it is. If mapmakers provide lots of Bouldered doors for the AI to face that's on them, and the for the player the balance seems ok to me. Let me know if you disagree.